### PR TITLE
[ISSUE #9] Add http sink task

### PR DESCRIPTION
**Pull request by zhaohai1299002788**
add http sink task

### CODE REVIEW

File: connectors/aliyun/rocketmq-connect-http/README.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
+# rocketmq-connect-http
```
Comment (duhenglucky) [inline connectors/aliyun/rocketmq-connect-http/README.md:1]: @zhaohai1299002788 Would you like to add an introduction for this connector？ other connectors also have this problem

File: connectors/aliyun/rocketmq-connect-http/README.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+## rocketmq-connect-http 打包
+```
+mvn clean install -Dmaven.test.skip=true
+```
+
+## rocketmq-connect-http 启动
+
+* **http-sink-connector** 启动
```
Comment (duhenglucky) [inline connectors/aliyun/rocketmq-connect-http/README.md:10]: How do I verify that this connector has been successfully started and is working well?

File: connectors/aliyun/rocketmq-connect-http/README.md (hunk 1 above)
Comment (zhaohai1299002788) [reply connectors/aliyun/rocketmq-connect-http/README.md:1]: HTTP, MNS, dingtalk, and FC have all added descriptions

File: connectors/aliyun/rocketmq-connect-http/src/main/java/com/aliyun/rocketmq/connect/http/sink/HttpSinkConnector.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    @Override
+    public void resume() {
+
+    }
+
+    @Override
+    public List<KeyValue> taskConfigs(int maxTasks) {
+        List<KeyValue> keyValueList = new ArrayList<>(11);
+        KeyValue keyValue = new DefaultKeyValue();
+        keyValue.put(HttpConstant.URL_CONSTANT, url);
```
[the remark this answers is not in the document]
Comment (zhaohai1299002788) [reply connectors/aliyun/rocketmq-connect-http/src/main/java/com/aliyun/rocketmq/connect/http/sink/HttpSinkConnector.java:33]: delete duplicate code

File: connectors/aliyun/rocketmq-connect-http/src/main/java/com/aliyun/rocketmq/connect/http/sink/HttpSinkTask.java [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    @Override
+    public void resume() {
+
+    }
+
+    @Override
+    public void validate(KeyValue config) {
+        if (StringUtils.isBlank(config.getString("url"))) {
+            throw new RuntimeException("http required parameter is null !");
+        }
```
[the remark this answers is not in the document]
Comment (zhaohai1299002788) [reply connectors/aliyun/rocketmq-connect-http/src/main/java/com/aliyun/rocketmq/connect/http/sink/HttpSinkTask.java:51]: add check whether the URL can be accessed

File: connectors/aliyun/rocketmq-connect-http/src/main/java/com/aliyun/rocketmq/connect/http/sink/HttpSinkTask.java [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+    private String url;
+
+    @Override
+    public void put(List<ConnectRecord> sinkRecords) throws ConnectException {
+        try {
+            sinkRecords.forEach(connectRecord -> {
+                OkHttpUtils.builder()
+                        .url(url)
+                        .addParam(HttpConstant.DATA_CONSTANT, JSON.toJSONString(connectRecord.getData()))
+                        .post(true)
```
[the remark this answers is not in the document]
Comment (zhaohai1299002788) [reply connectors/aliyun/rocketmq-connect-http/src/main/java/com/aliyun/rocketmq/connect/http/sink/HttpSinkTask.java]: Added trasform configuration.

**Comment by zhouxinyu** [comment]
HTTP connector doesn't belong to any vendor, should move from `aliyun` folder.

File: connectors/aliyun/rocketmq-connect-http/src/main/java/org/apache/rocketmq/connect/http/sink/HttpSinkConnector.java [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+    }
+
+    @Override
+    public void validate(KeyValue config) {
+        if (StringUtils.isBlank(config.getString(HttpConstant.URL_CONSTANT))) {
+            throw new RuntimeException("http required parameter is null !");
+        }
+        try {
+            URL urlConnect = new URL(url);
+            URLConnection urlConnection = urlConnect.openConnection();
```
Comment (2011shenlin) [inline connectors/aliyun/rocketmq-connect-http/src/main/java/org/apache/rocketmq/connect/http/sink/HttpSinkConnector.java:50]: validate is after the init(), so the url is null

File: connectors/aliyun/rocketmq-connect-http/src/main/java/org/apache/rocketmq/connect/http/sink/HttpSinkTask.java [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+
+    private String url;
+
+    @Override
+    public void put(List<ConnectRecord> sinkRecords) throws ConnectException {
+        try {
+            sinkRecords.forEach(connectRecord -> {
+                OkHttpUtils.builder()
+                        .url(url)
+                        .addParam(HttpConstant.DATA_CONSTANT, JSON.toJSONString(connectRecord.getData()))
```
Comment (2011shenlin) [inline connectors/aliyun/rocketmq-connect-http/src/main/java/org/apache/rocketmq/connect/http/sink/HttpSinkTask.java:28]: 直接connectRecord.getData().toString() 就可以了，不要做JOSN序列化，其他的connector一并处理下

**Comment by zhaohai1299002788** [comment]
> HTTP 连接器不属于任何供应商，应从`aliyun`文件夹中移出。

Moved out of aliyun folder.

File: connectors/aliyun/rocketmq-connect-http/src/main/java/org/apache/rocketmq/connect/http/sink/HttpSinkTask.java (hunk 7 above)
Comment (zhaohai1299002788) [reply connectors/aliyun/rocketmq-connect-http/src/main/java/org/apache/rocketmq/connect/http/sink/HttpSinkTask.java]: 已处理

File: connectors/aliyun/rocketmq-connect-http/src/main/java/org/apache/rocketmq/connect/http/sink/HttpSinkConnector.java (hunk 6 above)
Comment (zhaohai1299002788) [reply connectors/aliyun/rocketmq-connect-http/src/main/java/org/apache/rocketmq/connect/http/sink/HttpSinkConnector.java]: 已处理

**Pull request merged**